lang/LangPrimSource/PyrStringPrim.cpp

```diff
@@ -410,7 +410,7 @@ static int prString_FindRegexp(struct VMGlobals *g, int numArgsPushed)
 	int match_count = matches.size();
 
 	PyrObject *result_array = newPyrArray(g->gc, match_count, 0, true);
-	SetObject(a, result_array);
+	++g->sp; SetObject(g->sp, result_array); // push result to make reachable
```

Comment:
Below this, we create new objects, so result_array needs to be reachable to ensure it isn't collected. If overwrite the receiver it may no longer be reachable, so it could be collected. But we're still using its data below, so that can lead to the same sort of error we saw with flop.